dali/pipeline/operator/sequence_shape.h

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATOR_SEQUENCE_SHAPE_H_
+#define DALI_PIPELINE_OPERATOR_SEQUENCE_SHAPE_H_
+
+#include <string>
+#include <type_traits>
+#include <vector>
+#include "dali/pipeline/data/tensor.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+struct LayoutDesc {
+  static inline LayoutDesc FrameAndChannel(const TensorLayout &layout) {
+    int frame_dim = VideoLayoutInfo::FrameDimIndex(layout);
+    int channel_dim = ImageLayoutInfo::ChannelDimIndex(layout);
+    frame_dim = frame_dim > 1 ? -1 : frame_dim;
+    channel_dim = channel_dim > 1 ? -1 : channel_dim;
+    if (frame_dim != 0 && channel_dim != 0) {
+      return {};
+    }
+    return {layout, frame_dim, channel_dim};
+  }
```

Comment:
I find it quite disturbing that LayoutDesc does actually make decisions about the interpretation of the layout. I think this should be a job of ExpandDesc. LayoutDesc seems a bit superfluous.

Reply:
Dropped the LayoutDesc completely.

Reply:
:+1: